tests/e2e/specs/out-of-game/profile.spec.js

```diff
@@ -0,0 +1,344 @@
+import { myUser, playerOne, playerTwo } from '../../fixtures/userFixtures';
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import GameStatus from '../../../../utils/GameStatus.json';
+
+dayjs.extend(utc);
+
+describe('Profile Page', () => {
+  beforeEach(() => {
+    cy.wipeDatabase();
+    cy.visit('/');
+  });
+
+  describe('Profile page content', function() {
+    beforeEach(() => {
+      cy.signupPlayer(myUser);
+    });
+
+    it('Displays username', function() {
+      cy.vueRoute('/my-profile');
+      cy.contains('h1', 'My Profile').should('be.visible');
+      cy.get('[data-cy=username]').should('contain', myUser.username);
+    });
+
+    it('Shows Discord section when not connected', () => {
+      cy.vueRoute('/my-profile');
+      cy.contains('h2', 'Discord');
+      cy.get('[data-cy=not-connected]').should('be.visible');
+    });
+
+    it('Shows Discord connected state', function() {
+      cy.vueRoute('/my-profile');
+
+      cy.window().its('cuttle.authStore')
+        .then((authStore) => {
+          authStore.$patch({
+            identities: [
+              {
+                provider: 'discord',
+                username: 'TestDiscordUser#1234'
+              }
+            ]
+          });
+        });
+
+      cy.contains('h2', 'Discord').should('be.visible');
+      cy.get('[data-cy=discord-username]')
+        .should('be.visible')
+        .should('contain', 'TestDiscordUser#1234');
+      cy.get('[data-cy=not-connected]').should('not.exist');
+    });
+
+    it('Shows fallback when no games', function() {
+      cy.vueRoute('/my-profile');
+      cy.contains('h1', 'My Profile').should('be.visible');
+      cy.contains('No games found').should('be.visible');
+    });
+  });
+
+
+
+  describe('Profile Page - Scrolling and game list tests', () => {
+    beforeEach(() => {
+      cy.signupOpponent(myUser);
+      cy.signupOpponent(playerOne);
+    });
+
+    it('Loads more games on scroll', function() {
+      const games = Array.from({ length: 30 }).map((_, i) => ({
+        name: `Game ${i + 1}`,
+        status: GameStatus.FINISHED,
+        isRanked: true,
+        createdAt: dayjs.utc().subtract(i, 'day')
+          .toDate(),
+        p0: this[`${playerOne.username}Id`],
+        p1: this[`${myUser.username}Id`],
+        winner: this[`${myUser.username}Id`],
+      }));
+
+      cy.loadFinishedGameFixtures(games);
+      cy.loginPlayer(myUser);
+      cy.vueRoute('/my-profile');
+
+      cy.get('[data-test="game-list-item"]').should('have.length.at.least', 5);
+
+      cy.get('.v-virtual-scroll').scrollTo('bottom');
+      cy.wait(500);
+      cy.get('.v-virtual-scroll').scrollTo('bottom');
+
+      cy.window().then(win => {
+        const pinia = win.cuttle.app.config.globalProperties.$pinia;
+        const myGamesStore = pinia._s.get('myGames');
+        expect(myGamesStore.games.length).to.eq(30);
+      });
+
+      cy.contains('[data-test="game-list-item"]', 'Game 30').should('exist');
```

Comment:
let's go a step further and assert it's visible
```suggestion
      cy.contains('[data-test="game-list-item"]', 'Game 30').should('be.visible');
```

Reply:
Based on my comment above, I think we could move this line up and replace the cy.wait() call with this